arch/rv32i/src/machine_timer.rs

```diff
@@ -71,6 +72,8 @@ impl<'a> hil::time::Alarm<'a> for MachineTimer<'a> {
     }
 
     fn set_alarm(&self, tics: u32) {
+        // TODO: this may generate spurious interrupts due to it taking 2 instructions to write 64
+        // bit registers, see RISC-V Priveleged ISA
```

Comment:
Should we disable timer interrupts before setting this then?

Reply:
The correct approach involves setting the upper register of the timer to the max value before setting the lower register. I added a comment to describe this, but think that change belongs in its own PR.